cnd/src/network/mod.rs

```diff
@@ -128,6 +128,36 @@ impl Swarm {
             local_peer_id,
         })
     }
+
+    /// Takes all the data we know about a 'created' swap.  Executes the
+    /// appropriate communication protocols in order to send data to the peer
+    /// and receive data from the peer (remaining swap parameters) in order to
+    /// finalize this swap.
+    pub fn finalize_created_swap<A, B>(
+        &mut self,
+        swap_id: SwapId,
+        swap_digest: SwapDigest,
+        secret_hash: SecretHash,
+        alpha_identity: A,
+        beta_identity: B,
+    ) -> anyhow::Result<()> {
+        let mut swarm = self.swarm.lock().await;
+        // Assume new NetworkBehaviour is called comit still i.e., the
+        // NetworkBehaviour that implements the new communication protocols.
+
+        // Assumptions:
+        // - This works for Han-Han, Han-HALight, HALight-Han
+
+        // Questions:
+        // - Is this all the data we need for a Han-Herc20 swap?
```

Comment:
To answer this, I would recommend a bottom-up thinking approach: Look at the protocols we have, think through all the use-cases that we want to achieve and write down what the data is that we are going to need to implement those.

Reply:
Good tip, thanks.  Please see new version (force pushed, it is a total rewrite).